### PR TITLE
2018 02 16 w fits

### PR DESCRIPTION
**Pull request by sabrandt**

### CODE REVIEW

**Comment by sabrandt** [comment]
This PR contains the most recent versions of the signal extraction fitting macros.

File: SignalExtraction/fitWm_lumis_2d.C [hunk 1]
```diff
@@ -0,0 +1,2950 @@
...
+  // ===================== Recoil correction files ============================
+  
+  const TString directory("/afs/cern.ch/user/d/dalfonso/public/WZ/JULY5");
+  // for Puppi, inclusive
+  RecoilCorrector *recoilCorr = new  RecoilCorrector("","");
+  recoilCorr->loadRooWorkspacesMCtoCorrect(Form("%s/WmpMCPuppi_genFix/",directory.Data()));
+  // recoilCorr->loadRooWorkspacesMCtoCorrect(Form("%s/WmpMCPuppi/",directory.Data()));
+  // recoilCorr->loadRooWorkspacesData(Form("%s/ZeeDataPuppi_bkgTopEWK_par%s/",directory.Data(),input_section.Data()));
+  recoilCorr->loadRooWorkspacesData(Form("%s/ZmmDataPuppi_bkgTopEWK/",directory.Data()));
+  recoilCorr->loadRooWorkspacesMC(Form("%s/ZeeMCPuppi/",directory.Data()));
```
Comment (mariadalfonso) [inline SignalExtraction/fitWm_lumis_2d.C]: Use consistently the mm correction and not mix of Zee and Zmm recoilCorr->loadRooWorkspacesMC(Form("%s/ZmmMCPuppi/",directory.Data()));

File: SignalExtraction/fitWm_lumis_2d.C [hunk 2]
```diff
@@ -0,0 +1,2950 @@
...
+  // recoilCorr->loadRooWorkspacesMC(Form("%s/ZmmMCPuppi/",directory.Data()));
+  
+  RecoilCorrector *recoilCorrm = new  RecoilCorrector("","");
+  recoilCorrm->loadRooWorkspacesMCtoCorrect(Form("%s/WmmMCPuppi_genFix/",directory.Data()));
+  // recoilCorrm->loadRooWorkspacesMCtoCorrect(Form("%s/WmmMCPuppi/",directory.Data()));
+  // recoilCorrm->loadRooWorkspacesData(Form("%s/ZeeDataPuppi_bkgTopEWK_par%s/",directory.Data(),input_section.Data()));
+  recoilCorrm->loadRooWorkspacesData(Form("%s/ZmmDataPuppi_bkgTopEWK/",directory.Data()));
+  recoilCorrm->loadRooWorkspacesMC(Form("%s/ZeeMCPuppi/",directory.Data()));
```
Comment (mariadalfonso) [inline SignalExtraction/fitWm_lumis_2d.C]: Use consistently the mm correction and not mix of Zee and Zmm recoilCorr->loadRooWorkspacesMC(Form("%s/ZmmMCPuppi/",directory.Data()));

File: SignalExtraction/fitWm_lumis_2d.C [hunk 3]
```diff
@@ -0,0 +1,2950 @@
...
+          weight *= scale1fbDown*lumi*corr;
+          weight2 *= scale1fbDown*lumi2*corr;
+        } else {
+          weight2*=scale1fb*lumi2*corr;
+          weight *= scale1fb*lumi*corr;
+        }
+        
+		// std::cout << "after fill " << std::endl; 
+        
+        // Pre
```
Comment (mariadalfonso) [inline SignalExtraction/fitWm_lumis_2d.C:1199]: need to have the Minos() Options for meaningfull errors

File: SignalExtraction/fitWm_lumis_2d.C [hunk 4]
```diff
@@ -0,0 +1,2950 @@
...
+// fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/Select10MedID_newBacon_fixGen_2018_01_12/Lumi10Parts_Part"+input_section+"/Wmunu/ntuples/wm_select.root"); typev.push_back(eWmunu);
+	// fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/NewBacon_MediumEleID/Wmunu/ntuples/wm_select.raw.root");   typev.push_back(eWmunu);
+  // fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/NewBacon_MediumEleID/Wmunu/ntuples/boson_select.root");  typev.push_back(eBKG);
+  // fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/NewBacon_MediumEleID/Wmunu/ntuples/ewk_select1.root");  typev.push_back(eEWK);
+  
+  // original ntuples without EWK combined
+    fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/NewBacon_MediumEleID/Wmunu/ntuples/top_select.raw.root"); typev.push_back(eTtb);
+	fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/NewBacon_MediumEleID/Wmunu/ntuples/wx_select.raw.root"); typev.push_back(eWx);
```
Comment (mariadalfonso) [inline SignalExtraction/fitWm_lumis_2d.C]: These are nutples from the may10, there was a fix for the genBoson of the 27 october
/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/NewBacon_MediumEleID/Wmunu/ntuples/

use the wx,zxx,wm from the folder "2017_10_27_fixGen" and the ww,wz,zz from the "NewBacon_MediumEleID"

fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/2017_10_27_fixGen/Wmunu/ntuples/wm_select.raw.root");   typev.push_back(eWmunu);
 
fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/2017_10_27_fixGen/Wmunu/ntuples/wx_select.raw.root");  typev.push_back(eWx);
 
fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/2017_10_27_fixGen/Wmunu/ntuples/zxx_select.raw.root");  typev.push_back(eZxx);

File: SignalExtraction/fitWe_lumis_2d.C [hunk 5]
```diff
@@ -0,0 +1,2709 @@
...
+  recoilCorr->loadRooWorkspacesMC(Form("%s/ZeeMCPuppi/",directory.Data()));
+  // recoilCorr->loadRooWorkspacesMC(Form("%s/ZmmMCPuppi/",directory.Data()));
+  
+  RecoilCorrector *recoilCorrm = new  RecoilCorrector("","");
+  // recoilCorrm->loadRooWorkspacesMCtoCorrect(Form("%s/WmmMCPuppi/",directory.Data()));
+  recoilCorrm->loadRooWorkspacesMCtoCorrect(Form("%s/WmmMCPuppi_genFix/",directory.Data()));
+  recoilCorrm->loadRooWorkspacesData(Form("%s/ZeeDataPuppi_bkgTopEWK/",directory.Data()));
+  // recoilCorrm->loadRooWorkspacesData(Form("%s/ZmmDataPuppi_bkgTopEWK/",directory.Data()));
+  recoilCorrm->loadRooWorkspacesMC(Form("%s/ZeeMCPuppi/",directory.Data()));
+  // recoilCorrm->loadRooWorkspacesMC(Form("%s/ZmmMCPuppi/",directory.Data()));
```
Comment (mariadalfonso) [inline SignalExtraction/fitWe_lumis_2d.C]: should use consistently ZmmMCPuppi and not ZeeMCPuppi

File: SignalExtraction/fitWe_lumis_2d.C [hunk 6]
```diff
@@ -0,0 +1,2709 @@
...
+  // fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/NewBacon_MediumEleID/Wenu/ntuples/we_select.root");   typev.push_back(eWenu);
+  // // fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/NewBacon_MediumEleID/Wenu/ntuples/ewk_select1.root");  typev.push_back(eEWK);
+  // fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/NewBacon_MediumEleID/Wenu/ntuples/boson_select.root");  typev.push_back(eBKG);
+
+  // // the samples with inclusive PU RW
+    // fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/2017_10_27_fixGen/Wenu/ntuples/we_select.root");   typev.push_back(eWenu);
+  // fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/NewBacon_MediumEleID/Wenu/ntuples/ewk_select1.root");  typev.push_back(eEWK);
+  // fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/2017_10_27_fixGen/Wenu/ntuples/boson_select.root");  typev.push_back(eBKG);
+  
+   fnamev.push_back("/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/Select10MedID_newBacon/Lumi10Parts_Part"+input_section+"/AntiWenu/ntuples/data_select.raw.root"); typev.push_back(eAntiData);
```
Comment (mariadalfonso) [inline SignalExtraction/fitWe_lumis_2d.C]: why not using this one 
"/eos/cms/store/user/sabrandt/StandardModel/FlatNtuples/Select10MedID_newBacon_fixGen_2018_01_12
the one you are using is from the sept7 the resolution affect the data ?

File: SignalExtraction/fitWe_lumis_2d.C [hunk 7]
```diff
@@ -0,0 +1,2709 @@
...
+              recoilCorrm1->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep_raw->Pt(),lep->Phi(),pU1,pU2,0,0,0,doKeys); 
+           }          
+          TVector2 vMetCorr((corrMet)*cos(corrMetPhi),(corrMet)*sin(corrMetPhi));
+          Double_t corrMetWithLepton = (vMetCorr + vLepRaw - vLepCor).Mod();
+          if(typev[ifile]==eWenu)
+          {             
+            hWenuMet->Fill(corrMetWithLepton,weight);
+            hWenuMetm->Fill(corrMetWithLepton,weight*w2); corrMet=met, corrMetPhi=metPhi;
+            hWenuMetm_PileupUp->Fill(corrMetWithLepton,weightUp); corrMet=met, corrMetPhi=metPhi;
+            hWenuMetm_PileupDown->Fill(corrMetWithLepton,weight
```
Comment (mariadalfonso) [inline SignalExtraction/fitWe_lumis_2d.C]: need to use Minos() routine for meaningful errors

**Pull request closed without merging**